app/js/lib/speech/synthesis.js

```
@@ -0,0 +1,74 @@
+'use strict';
+
+const p = Object.freeze({
+  // Properties
+  synthesis: Symbol('synthesis'),
+  supportsSynthesis: Symbol('supportsSynthesis'),
+  voice: Symbol('voice'),
+
+  // Methods
+  populateVoice: Symbol('populateVoice'),
+});
+
+export default class SpeechSynthesis {
+  constructor() {
+    const synthesis = window.speechSynthesis;
+
+    this[p.supportsSynthesis] = !!synthesis;
+
+    if (this[p.supportsSynthesis]) {
+      this[p.synthesis] = synthesis;
+      this[p.synthesis].onvoiceschanged = this[p.populateVoice].bind(this);
+    } else {
+      this[p.synthesis] = null;
+    }
+
+    this[p.voice] = null;
+
+    Object.seal(this);
+  }
+
+  /**
+   * Speak a text aloud.
+   *
+   * @param {string} text
+   */
+  speak(text = '') {
+    if (!text) {
```

To me it would be an error case, it doesn't seem to make sense to call speak with an empty string ever.  I'd be inclined to fail early and throw.